File: frontend/src/api/dto/windDirection.ts
export type WindDirection = (typeof WindDirection)[keyof typeof WindDirection]

export const WindDirection = {
  NORTH: 'NORTH',
  NORTH_EAST: 'NORTH_EAST',
  EAST: 'EAST',
  SOUTH_EAST: 'SOUTH_EAST',
  SOUTH: 'SOUTH',
  SOUTH_WEST: 'SOUTH_WEST',
  WEST: 'WEST',
  NORTH_WEST: 'NORTH_WEST',
} as const

File: backend/src/main/java/fr/pedalons/enums/WindDirection.java
package fr.pedalons.enums;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public enum WindDirection {
  NORTH(0),
  NORTH_EAST(45),
  EAST(90),
  SOUTH_EAST(135),
  SOUTH(180),
  SOUTH_WEST(225),
  WEST(270),
  NORTH_WEST(315);

  final int angle;
}
